core/src/main/scala/cats/mtl/ApplicativeHandle.scala

```diff
@@ -1,34 +1,177 @@
 
-  def handle[A](fa: F[A])(f: E => A): F[A] =
-    handleWith(fa)(e => applicative.pure(f(e)))
+      val functor: Functor[EitherC[E]#l] = cats.instances.either.catsStdInstancesForEither
+
+      def raise[A](e: E): Either[E, A] = Left(e)
+
+      def handleWith[A](fa: Either[E, A])(f: E => Either[E, A]): Either[E, A] = fa match {
+        case Left(e) => f(e)
+        case r @ Right(_) => r
+      }
+    }
+  }
+
+  implicit final def handleOptionT[M[_]](implicit M: Monad[M]): ApplicativeHandle[OptionTC[M]#l, Unit] = {
+    new ApplicativeHandle[OptionTC[M]#l, Unit] {
+      val applicative: Applicative[OptionTC[M]#l] = OptionT.catsDataMonadForOptionT(M)
+
+      val functor: Functor[OptionTC[M]#l] = OptionT.catsDataFunctorForOptionT(M)
+
+      def raise[A](e: Unit): OptionT[M, A] = OptionT(M.pure(None))
+
+      def handleWith[A](fa: OptionT[M, A])(f: Unit => OptionT[M, A]): OptionT[M, A] =
+        OptionT(M.flatMap(fa.value) {
+          case None => f(()).value
+          case s @ Some(_) => M.pure(s)
+        })
+    }
+  }
+
+  implicit final def handleOption[E]: ApplicativeHandle[Option, Unit] = {
+    new ApplicativeHandle[Option, Unit] {
+      val applicative: Applicative[Option] = cats.instances.option.catsStdInstancesForOption
+
+      val functor: Functor[Option] = cats.instances.option.catsStdInstancesForOption
+
+      def raise[A](e: Unit): Option[A] = None
+
+      def handleWith[A](fa: Option[A])(f: Unit => Option[A]): Option[A] = fa match {
+        case None => f(())
+        case s @ Some(_) => s
+      }
+    }
+  }
+
+  implicit final def handleValidated[E](implicit E: Semigroup[E]): ApplicativeHandle[Validated[E, ?], E] =
+    new ApplicativeHandle[Validated[E, ?], E] {
+      val applicative: Applicative[Validated[E, ?]] = Validated.catsDataApplicativeErrorForValidated[E]
+
+      val functor: Functor[Validated[E, ?]] = Validated.catsDataApplicativeErrorForValidated[E]
+
+      def raise[A](e: E): Validated[E, A] = Validated.Invalid(e)
+
+      def handleWith[A](fa: Validated[E, A])(f: E => Validated[E, A]): Validated[E, A] = fa match {
+        case Validated.Invalid(e) => f(e)
+        case v @ Validated.Valid(_) => v
+      }
+    }
+
+  implicit final def handleIorT[F[_], E](implicit E: Semigroup[E], F: Monad[F]): ApplicativeHandle[IorT[F, E, *], E] =
+    new ApplicativeHandle[IorT[F, E, *], E] {
+      val applicative: Applicative[IorT[F, E, ?]] = IorT.catsDataMonadErrorForIorT[F, E]
+
+      val functor: Functor[IorT[F, E, ?]] = IorT.catsDataMonadErrorForIorT[F, E]
+
+      def raise[A](e: E): IorT[F, E, A] = IorT.leftT(e)
+
+      def handleWith[A](fa: IorT[F, E, A])(f: E => IorT[F, E, A]): IorT[F, E, A] = IorT(F.flatMap(fa.value) {
+        case Ior.Left(e) => f(e).value
+        case e @ _ => F.pure(e)
+      })
+    }
+  
+  implicit final def handleIor[E](implicit E: Semigroup[E]): ApplicativeHandle[Ior[E, *], E] =
```

Comment:
the naming for these instances seems to be quite inconsistent:
there is handleIor,  applicativeHandleKleisli, and applicativeLocalForKleisli.

Reply:
Would definitely like to be more consistent here. Should probably follow the same convention as cats-core.